src/main/java/com/mercadolibrewannabe/model/User.java

```diff
@@ -21,6 +26,9 @@
 	@Size(min = 6)
 	private String password;
 
+	@ManyToMany
+	private Set<Role> roles;
```

Comment:
Faltou o equals e hashcode lá na Role. Quer dizer, eu pelo menos não vi :).

Reply:
Aqui também é bom já deixar o valor inicializado :). Se alguém instanciar um User na mão e for operar sobre as roles vai tomar NPE

Reply:
> Faltou o equals e hashcode lá na Role. Quer dizer, eu pelo menos não vi :)

Incluído!

---

Como não faz parte da atividade, acabei nem implementando isso. Isso faz parte da experimentação também junto ao Spring Security. Ao meu ver seria melhor restringir a instancialização no construtor mesmo, o que acha? 

Ao enviar um UserForm já vou ter que mandar qual seria o Id da minha `Role` também. Mas enfim, faria isso depois, já que quero focar mais nas atividades da jornada, o que me diz?

Reply:
Eita João, me perdoe. Acho que não entendi aqui seu comentário. Pode me explicar de novo com algum exemplo? Foi mal.

Reply:
Que isso, tranquilo! Seguinte, pensei em instanciar o conjunto de `roles` somente no construtor mesmo. Do mesmo jeito que que existe a criação do e-mail e da senha.

E por enquanto, no sistema isso só é feito a partir do `form` que chama o construtor da classe principal no `toModel()`.

Daí, iria acrescentar no parâmetro do toModel um RoleRepository, ou uma function que recebesse um Id e retornasse uma Role ou uma interface que o RoleRepository implementasse.

Dessa forma o form iria ter o(s) Id(s) da Role e eu poderia instanciar dessa maneira. Ficou mais claro?

Reply:
Ah, tudo bem! Geralmente no cadastro inicial de um usuário as roles são definidas de maneira automática, justamente para alguém não passar um id diferente e ganhar super poderes. Geralmente a role de 1 é a de admin :P.